central-service/controllers/room.go

```diff
@@ -78,19 +78,73 @@ func (controller *RoomController) LeaveRoomEndpoint(c echo.Context) error {
+		if err, ok := err.(services.BSGError); ok {
+			return echo.NewHTTPError(err.StatusCode, "Failed to leave room. "+err.Error())
 		}
-		return echo.NewHTTPError(http.StatusInternalServerError)
+		return echo.NewHTTPError(http.StatusInternalServerError, "Failed to leave room. Please try again later")
 	}
 	return c.JSON(http.StatusOK, map[string]string{
 		"message": "Successfully Left Room",
 	})
 }
 
+func (controller *RoomController) CreateNewRoundEndpoint(c echo.Context) error {
+	var roundCreationParams services.RoundCreationParameters
+	if err := c.Bind(&roundCreationParams); err != nil {
+		return echo.NewHTTPError(http.StatusBadRequest, "Invalid data. Please try again")
+	}
+	roomID := c.Param("roomID")
+	newRound, err := controller.roomService.CreateRound(&roundCreationParams, roomID)
+	if err != nil {
+		log.Printf("Failed to create new round: %v\n", err)
+		if err, ok := err.(*services.BSGError); ok {
+			return echo.NewHTTPError(err.StatusCode, "Failed to create round. "+err.Message)
+		}
+		return echo.NewHTTPError(http.StatusInternalServerError, "Failed to create round. Please try again later")
+	}
+	return c.JSON(http.StatusCreated, map[string]models.Round{
+		"data": *newRound,
+	})
+}
+
+func (controller *RoomController) StartRoundEndpoint(c echo.Context) error {
+	targetRoomID := c.Param("roomID")
+	userAuthID := c.Get("authToken").(*auth.Token).UID
+	roundStartTime, err := controller.roomService.StartRoundByRoomID(targetRoomID, userAuthID)
+	if err != nil {
+		log.Printf("Failed to start round for room with id %s: %v\n", targetRoomID, err)
+		if err, ok := err.(*services.BSGError); ok {
+			return echo.NewHTTPError(err.StatusCode, "Failed to start round. "+err.Message)
+		}
+		return echo.NewHTTPError(http.StatusInternalServerError, "Failed to start round. Please try again later")
+	}
+	return c.JSON(http.StatusOK, map[string]interface{}{
+		"startTime": roundStartTime.Unix(),
+	})
+}
+
+func (controller *RoomController) GetLeaderboardEndpoint(c echo.Context) error {
+	roomID := c.Param("roomID")
+	leaderboard, err := controller.roomService.GetLeaderboard(roomID)
+	if err != nil {
+		log.Printf("Failed to get leaderboard for room with id %s: %v\n", roomID, err)
+		if err, ok := err.(services.BSGError); ok {
+			return echo.NewHTTPError(err.StatusCode, "Failed to get leaderboard. "+err.Message)
+		}
+		return echo.NewHTTPError(http.StatusInternalServerError, "Failed to get leaderboard. Please try again later")
+	}
+	return c.JSON(http.StatusOK, map[string]interface{}{
+		"leaderboard": leaderboard,
+	})
+}
+
 func (controller *RoomController) InitializeRoutes(g *echo.Group) {
 	g.POST("/", controller.CreateNewRoomEndpoint)
-	g.GET("/:roomID", controller.FindRoomEndpoint)
 	g.POST("/:roomID/join", controller.JoinRoomEndpoint)
 	g.POST("/:roomID/leave", controller.LeaveRoomEndpoint)
-}
\ No newline at end of file
+	g.POST("/:roomID/new-round", controller.CreateNewRoundEndpoint)
```

Comment:
nit: can we get rid of the `-` somehow? Like rewrite the endpoint as something like `/:roomID/rounds/create`